Tactics/Simp.lean

```diff
@@ -52,12 +52,18 @@ def LNSymSimpContext
   (exprs : Array Expr := #[])
   -- Simprocs to add to the default set.
   (simprocs : Array Name := #[])
+  -- Whether the default simprocs should be used.
+  (useDefaultSimprocs : Bool := true)
   -- argument to `DiscrTree.mkPath`
   (noIndexAtArgs : Bool := true)
   : MetaM (Simp.Context ×  Array Simp.Simprocs) := do
   let mut ext_simpTheorems := #[]
-  let default_simprocs ← Simp.getSimprocs
-  let mut all_simprocs := (#[default_simprocs] : Simp.SimprocsArray)
+  /- Workaround for https://github.com/leanprover/lean4/issues/5607: Elaboration failure with let mut whose RHS is a do notation -/
+  let all_simprocs ← do
+    if useDefaultSimprocs then
+      pure #[← Simp.getSimprocs]
+    else pure #[]
+  let mut all_simprocs := all_simprocs
```

Comment:
This also works (although the use of `:=` and `←` looks a bit wonky). I'll let you pick which one you find cleaner
```suggestion
  let mut all_simprocs :=← do
    if useDefaultSimprocs then
      pure #[← Simp.getSimprocs]
    else pure #[]
```

Reply:
I'll pick the current one.